zcash_client_sqlite/src/lib.rs

```diff
@@ -337,11 +338,16 @@ impl<C: Borrow<rusqlite::Connection>, P: consensus::Parameters> WalletRead for W
         }
     }
 
-    fn is_seed_relevant_to_any_derived_accounts(
+    fn seed_relevance_to_derived_accounts(
         &self,
         seed: &SecretVec<u8>,
-    ) -> Result<bool, Self::Error> {
+    ) -> Result<SeedRelevance<Self::AccountId>, Self::Error> {
+        let mut has_accounts = false;
+        let mut has_derived = false;
+        let mut relevant_account_ids: Option<NonEmpty<Self::AccountId>> = None;
```

Comment:
nonblocking: Instead of this, just make it a `Vec`, and then at the end use `NonEmpty::from_vec`.